pypuf/tools.py

```diff
@@ -260,14 +260,23 @@ class TrainingSet():
     Note that this is, strictly speaking, not a set.
     """
 
-    def __init__(self, instance, N):
+    def __init__(self, instance, N, random_instance=RandomState(), reps=None):
         """
         :param instance: pypuf.simulation.base.Simulation
                          Instance which is used to generate responses for random challenges.
         :param N: int
                   Number of desired challenges
         """
         self.instance = instance
-        self.challenges = array(list(sample_inputs(instance.n, N)))
-        self.responses = instance.eval(self.challenges)
+        self.challenges = array(list(sample_inputs(instance.n, N, random_instance=random_instance)))
+        if reps is None:
+            reps = 1
+        self.responses = zeros((reps, N))
+        for i in range(reps):
+            self.challenges, cs = itertools.tee(self.challenges)
```

Comment:
why are use assigning a value to `self.challenges` here? I think it is unused and later overwritten. (Use `_` instead?)